docs/ci-cd-environment/setup-caching-on-aws-s3.md

```diff
@@ -0,0 +1,164 @@
+      ]
+    }
+  ]
+}
+EOF
+
+aws iam create-policy \
+  --policy-name semaphore-cache-policy \
+  --policy-document file:///tmp/semaphore-cache-policy.json \
+  --description "Restricts access to some operations on the semaphore-cache S3 bucket"
+```
+
+The `aws iam create-policy` command will generate an output like this:
+
+```json
+{
+  "Policy": {
+    "PolicyName": "semaphore-cache-policy",
+    "CreateDate": "2015-06-01T19:31:18.620Z",
+    "AttachmentCount": 0,
+    "IsAttachable": true,
+    "PolicyId": "ZXR6A36LTYANPAI7NJ5UV",
+    "DefaultVersionId": "v1",
+    "Path": "/",
+    "Arn": "arn:aws:iam::0123456789012:policy/semaphore-cache-policy",
+    "UpdateDate": "2015-06-01T19:31:18.620Z"
+  }
+}
+```
+
+Make sure you save the `Arn` of the AWS IAM policy. You'll use that value when attaching the AWS IAM policy to the AWS IAM user.
+
+<b>3. Create the AWS IAM user and its access keys:</b>
+
+This is the AWS IAM user the cache CLI will use to authorize and get access to your AWS bucket:
+
+```
+aws iam create-user --user-name semaphore
+aws iam create-access-key --user-name semaphore
+```
+
+The `aws iam create-access-key` command will generate an output like this:
+
+```json
+{
+  "AccessKey": {
+    "UserName": "semaphore",
+    "Status": "Active",
+    "CreateDate": "...",
+    "SecretAccessKey": "...",
+    "AccessKeyId": "..."
+  }
+}
+```
+
+Make sure you save the `SecretAccessKey` and the `AccessKeyId` values. We will need them when configuring the `~/.aws` folder in the agent machine.
+
+<b>4. Attach the AWS IAM policy to the AWS IAM user</b>
+
+Let's attach the AWS IAM policy we created on step #2 to the user we created on step #3:
+
+```
+aws iam attach-user-policy \
+  --user-name semaphore \
+  --policy-arn "[this is where the policy ARN you saved on step #2 needs to go]"
+```
+
+## Configure the agent machine
+
+Once you have all the AWS resources created, let's jump into the machine running the agent and configure the AWS credentials for the cache CLI to use.
+
+<b>1. Configure the `~/.aws` folder</b>
+
+Make sure you use the `SecretAccessKey` and `AccessKeyId` for the proper AWS IAM user. Also, make sure you replace `us-east-1` with your region of choice and `semaphore-cache` with your bucket name.
+
+```
+mkdir ~/.aws
+
+cat > ~/.aws/config <<EOF
+[default]
+region = us-east-1
+output = json
+EOF
+
+cat > ~/.aws/credentials <<EOF
+[default]
+aws_access_key_id = SecretAccessKey
+aws_secret_access_key = AccessKeyId
+EOF
+```
+
+Note that the `~/.aws` folder should be created in the home directory of the same user running the agent.
+
+<b>2. Update the agent configuration</b>
+
+Finally, let's instruct the agent to pass the required environment variables to the cache CLI, by adding them in your agent's `config.yaml`:
+
+```yaml
+env-vars:
+  - SEMAPHORE_CACHE_BACKEND=s3
+  - SEMAPHORE_CACHE_S3_BUCKET=semaphore-cache
+```
+
+Restart your agent and you should be able to start using your S3 bucket as a cache storage.
```

Comment:
I imagine that whoever needs to follow this page will be neck-deep in sysadmin trial-and-error. I feel that this page lacks a tip on how to validate that everything is working as a relieving conclusion.

Reply:
It's a bit tricky though, because there's not a command that you run and check if things are working. You need to re-run your pipeline (if you already have one using the cache CLI) or create one to use, and I didn't want to go into that in this page.

Reply:
This is fine, I didn't expect that this would map to some definite instructions but more like a reminder "if all went well then when you re-run or launch a new pipeline things should work"